src/version.rs

```diff
@@ -185,6 +270,32 @@ impl Version {
                 build: vec![],
             })
     }
+
+    pub fn store_build_info(&self,
+                            conn: &PgConnection,
+                            info: upload::VersionBuildInfo) -> CargoResult<()> {
+
+        // Verify specified Rust version will parse before doing any inserting
+        info.channel_version()?;
+
+        let bi = NewBuildInfo {
+            version_id: self.id,
+            rust_version: info.rust_version,
+            target: info.target,
+            passed: info.passed,
+        };
+
+        diesel::insert(&bi.on_conflict(
+            build_info::table.primary_key(),
+            do_update()
+                .set((build_info::passed.eq(excluded(build_info::passed)),
+                      build_info::updated_at.eq(now_utc().to_timespec())))
```

Comment:
`updated_at` is handled by database triggers everywhere else. Should we do the same here? (adding `SELECT diesel_manage_updated_at('build_info');` to your migration will have Diesel set up the trigger automatically.